src/types/zkgyaml.ts

```diff
@@ -195,4 +195,35 @@ export class CLEYaml {
   toString() {
     return yaml.dump(this)
   }
+
+  decidePublishNetwork(): string | undefined {
+    // 1. if there's ethereum destination use that destination network
+    const ddkinds = this.dataDestinations.map((dataDest: { kind: any }) => { return dataDest.kind })
+    const ddethidx = ddkinds.indexOf('ethereum')
+    if (ddethidx >= 0) {
+      return (this.dataDestinations[ddethidx] as EthereumDataDestination).network
+    }
+    else {
+    // 2. if no eth dd, use DS ethereum network
+      const dskinds = this.dataSources.map((dataSrc: { kind: any; unsafe?: boolean }) => {
+        return dataSrc.kind + (dataSrc.unsafe ? '.unsafe' : '')
+      })
+
+      const dsethidx = dskinds.indexOf('ethereum')
+      if (dsethidx >= 0) {
+        return (this.dataSources[dsethidx] as EthereumDataSource).network
+      }
+      else {
+        // 3. if still nonen then use DS ethereum.unsafe network
+        const dsethunsafeidx = dskinds.indexOf('ethereum.unsafe')
+        if (dsethunsafeidx >= 0) {
+          return (this.dataSources[dsethunsafeidx] as EthereumDataSource).network
+        }
+        else {
+          // 4. else, no network can be used.
+          return undefined
```

Comment:
Should throw error here. Also this function is not used in `publish.ts` or `verify.ts` yet.

Reply:
Yea my original thought is: 1. if 4). then should use a user provided network, but you are right, can throw error to pass the info to caller
2. yes it's as expected, want to let the caller to call it, like in the tests, so that can keep the params clean (i.e. w/o cleYaml, only verifierAddress). what's you thoughts on this design?

Reply:
TODO: improve the logic here, we should decide which network before generating signer class, or just switch network in the frontend before calling the api.